torch/jit/__init__.py

```diff
@@ -1162,7 +1162,11 @@ def __setattr__(self, attr, value):
                         return
                 if isinstance(value, Attribute):
                     the_type = torch.jit.annotations.ann_to_type(value.type)
-                    self._register_attribute(attr, the_type, value.value)
+                    try:
+                        self._register_attribute(attr, the_type, value.value)
+                    except RuntimeError:
+                        raise RuntimeError("Could not register attribute '{}' of type '{}' for a value of type '{}'"
```

Comment:
why not expose original exception message?

Reply:
If the type passed doesn't match the one specified by the user it throws a pretty opaque pybind11 error, and there's no easy way to check that a type from the `typing` module matches the `type(some_value)`